game-app/game-core/src/main/java/games/strategy/triplea/ui/EditPanel.java

```diff
@@ -490,33 +494,40 @@ public void actionPerformed(final ActionEvent event) {
             setWidgetActivation();
             final PlayerChooser playerChooser =
                 new PlayerChooser(getData().getPlayerList(), getMap().getUiContext(), false);
+            final Container parent = getTopLevelAncestor();
             final JDialog dialog =
-                playerChooser.createDialog(getTopLevelAncestor(), "Select owner PUs to change");
+                playerChooser.createDialog(parent, "Select Owner PUs/Resources to Change");
             dialog.setVisible(true);
             final GamePlayer player = playerChooser.getSelected();
             if (player == null) {
               cancelEditAction.actionPerformed(null);
               return;
             }
-            getData().acquireReadLock();
-            final Resource pus;
-            try {
-              pus = getData().getResourceList().getResource(Constants.PUS);
-            } finally {
-              getData().releaseReadLock();
-            }
-            if (pus == null) {
-              cancelEditAction.actionPerformed(null);
-              return;
+
+            final List<Resource> resources = getData().getResourceList().getResources().stream()
+                    .filter(r -> !r.getName().equals(Constants.VPS))
+                    .collect(Collectors.toList());
+            final Resource resource;
+            if (resources.size() == 1) {
+              resource = resources.get(0);
+            } else {
+              final ResourceChooser chooser =
+                      new ResourceChooser(resources, getMap().getUiContext());
+              resource = chooser.showDialog(parent, "Select resource to change");
+              if (resource == null) {
+                cancelEditAction.actionPerformed(null);
+                return;
+              }
```

Comment:
We are getting into long method territory. Have you considered extracting the entirety of the edit resources dialog to it's own class? Seemingly edit panel should not be aware of how to implement every single edit function in its entirety.
Another topic, this is getting kinda dense, some guiding comments or breaking this up into helper methods would be very helpful and avoid a maintainer from having to reverse engineer the code to get an idea what is happening.

Another question, why is VPS excluded?

Reply:
I excluded VPS because it's also excluded in other parts of the code - e.g. the economy panel which shows the actual resources.

If you think they should be included, I can keep them.

Reply:
I'm actually not sure what VPS really even is!  = D
(Victory points??)
Hence the exclusion was interesting.